### PR TITLE
TST: avoid `NameError`s in a couple tests that couldn't be run in isolation

### LINKED ISSUE #17558: TST: two tests from  `astropy/io/registry/tests/test_registries.py` that cannot be run in isolation
This test cannot be run on its own at the moment:
```
$ pytest astropy/io/registry/tests/test_registries.py::TestUnifiedOutputRegistry::test_delay_doc_updates
...
FAILED astropy/io/registry/tests/test_registries.py::TestUnifiedOutputRegistry::test_delay_doc_updates - UnboundLocalError: cannot access local variable 'ihd' where it is not associated with a value
```
this test actually contains some logic to workaround test pollution, but it turns out it really *depends* on test pollution to run properly.

### PR DESCRIPTION
**Pull request by neutrinoceros**
### Description
Fixes #17558

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

File: astropy/io/registry/tests/test_registries.py [hunk 1]
```diff
@@ -353,6 +353,7 @@ def test_delay_doc_updates(self, registry, fmtcls1):
                     assert docs[-1][iread : iread + 3] != "Yes"
         # now test it's updated
         docs = EmptyData.read.__doc__.split("\n")
+        ihd = [i for i, s in enumerate(docs) if ("Format" in s)][0]
```
Comment (nstarman) [inline astropy/io/registry/tests/test_registries.py:356]: I think a slightly larger factor is called for here. The above test block could be dedented from the context and then some of the index creators can be moved out of the conditional block using the solution here.

Comment (neutrinoceros) [reply astropy/io/registry/tests/test_registries.py:356]: > The above test block could be dedented from the context

I'm not sure that's the case; without looking to deeply into how this context works exactly, it seems to me that the only difference between line 346 and line 355 is that the former happens within the context, while the latter doesn't.

Comment (nstarman) [reply astropy/io/registry/tests/test_registries.py:356]: Hm. you're right. It happens at context exit.

**Review by nstarman — APPROVED** [review]

**Comment by pllim** [comment]
I see that you applied "affects-dev" label but if it only affects development version, why the backport?

**Comment by pllim** [comment]
Anyways, I will remove that label. Thanks, all!

**Comment by neutrinoceros** [comment]
I guess I had a slightly wrong interpretation of that label: I thought it meant "doesn't affect end users". Anyway, thanks for fixing my mistake !

**Comment by pllim** [comment]
No worries. It was added years ago by `embray` for change log check in the sense that we do not need change log if a bug has not been released yet. Now I think it is mostly informational.

**Pull request merged**